File: ajps-backend/src/main/java/com/himusharier/ajps_backend/model/FileUpload.java
package com.himusharier.ajps_backend.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.himusharier.ajps_backend.constants.FileUploadOrigin;
import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@Builder
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "AJPS_FILES")
public class FileUpload {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private FileUploadOrigin fileOrigin;

    @Column(nullable = false)
    private String storedName;

    @Column(nullable = false)
    private String originalName;

    @Column(nullable = false)
    private long size;

    @Column(nullable = false)
    private String type;

    @Column(nullable = false)
    private String fileUrl;

    @Column(nullable = false)
    private boolean isReviewFile = false;

    @Column(nullable = false)
    private boolean isCopyEditingFile = false;

    @Column(nullable = false)
    private boolean isProductionFile = false;

    @Column(nullable = false)
    private boolean isPublicationFile = false;

    @ManyToOne
    @JoinColumn(name = "submission_id", nullable = false)
    @JsonBackReference
    private Submission submission;
}

File: ajps_flutter_app/lib/models/file_upload.dart
class FileUpload {
  final int id;
  final String fileOrigin;
  final String storedName;
  final String originalName;
  final int size;
  final String type;
  final String fileUrl;

  FileUpload({
    required this.id,
    required this.fileOrigin,
    required this.storedName,
    required this.originalName,
    required this.size,
    required this.type,
    required this.fileUrl,
  });

  factory FileUpload.fromJson(Map<String, dynamic> json) {
    return FileUpload(
      id: json['id'],
      fileOrigin: json['fileOrigin'],
      storedName: json['storedName'],
      originalName: json['originalName'],
      size: json['size'],
      type: json['type'],
      fileUrl: json['fileUrl'],
    );
  }
}
